File: algorithms/warmup/chocolate_feast.pl
use v5.10;

while (<>) {
    next unless $i++;
    my @c = split ' ', $_;
    my $c = $r = $c[0] / $c[1];
    my $m = $c[2];
    
    while ($c >= $m) {
        $r += int($c / $m);
        $c = int($c / $m) + ($c % $m);
    }
    say $r;
}

File: algorithms/warmup/chocolate_feast.py
T = int(raw_input())
for i in range(T):
    n,c,m = [int(x) for x in raw_input().split(' ')]
    t = w = n / c;
    while w >= m:
        r = w / m
        t += r
        w %= m
        w += r
    print t
